gaeguli/fifo-transmit.c

```diff
@@ -296,12 +301,12 @@ gaeguli_fifo_transmit_get_read_status (GaeguliFifoTransmit * self)
   return status;
 }
 
-GaeguliFifoTransmitStats *
+GVariantDict *
 gaeguli_fifo_transmit_get_stats (GaeguliFifoTransmit * self)
 {
   g_return_val_if_fail (GAEGULI_IS_FIFO_TRANSMIT (self), NULL);
 
-  return &self->stats;
+  return g_variant_dict_ref (self->stats);
```

Comment:
This might cause programmer error; in case that user calls unref till refcount reaches zero.
But for now, this can be acceptable because something like `g_variant_dict_dup()` doesn't exist in glib.
What do you think?

Reply:
Yes, but unref'ing the dict twice would be, as you say, programmer error.

How is this different from when I - for example - do `gst_bin_get_by_name()` and after I no longer need the GstBin reference I've obtained, I have to release it by calling `gst_object_unref()` once and only once? Not observing that requirement can crash my hypothetical GStreamer pipeline in a very similar way.

Reply:
That makes sense. 
What I am concerned about is that analytics logic would be run in a multithread environment.
but yes, it will be okay.